StyleCop.Analyzers/StyleCop.Analyzers.Test/NamingRules/SA1308UnitTests.cs

```diff
@@ -0,0 +1,82 @@
+﻿
+namespace StyleCop.Analyzers.Test.NamingRules
+{
+    using Microsoft.CodeAnalysis.Diagnostics;
+    using StyleCop.Analyzers.NamingRules;
+    using System;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using TestHelper;
+    using Xunit;
+
+    public class SA1308UnitTests : DiagnosticVerifier
+	{
+		private const string DiagnosticId = SA1308VariableNamesMustNotBePrefixed.DiagnosticId;
+
+		[Fact]
+		public async Task TestEmptySource()
+		{
+			var testCode = string.Empty;
+			await this.VerifyCSharpDiagnosticAsync(testCode, EmptyDiagnosticResults, CancellationToken.None);
+		}
+
+
+        private readonly string[] modifiers = new[] { "public", "private", "protected", "public readonly", "internal readonly", "public static", "private static" };
+       
+
+		[Fact]
+		public async Task TestFieldStartingWithPrefixesToTriggerDiagnostic()
+		{
+            foreach (var modifier in this.modifiers)
+            {
+                await this.TestFieldSpecifyingModifierAndPrefix(modifier, "m_", "m_");
+                await this.TestFieldSpecifyingModifierAndPrefix(modifier, "s_", "s_");
+                await this.TestFieldSpecifyingModifierAndPrefix(modifier, "m\u005F", "m_");
```

Comment:
:bulb: Interesting test, but I think you meant `\\u005F`. In it's current form, it's exactly equivalent to the first test.